packages/@angular/cli/commands/lint.ts

```diff
@@ -22,6 +23,12 @@ export default Command.extend({
       description: 'Fixes linting errors (may overwrite linted files).'
     },
     {
+      name: 'type-check',
+      type: Boolean,
+      default: false,
```

Comment:
It is, yes. I think it's justified because of the problem that having it on seems to be causing, but we should document it.